File: Java/Agenda/app/src/main/java/io/github/henriquemcc/agenda/java/ui/ListaAlunosView.java
package io.github.henriquemcc.agenda.java.ui;

import android.app.AlertDialog;
import android.view.MenuItem;
import android.widget.AdapterView;
import android.widget.ListView;

import androidx.annotation.NonNull;

import io.github.henriquemcc.agenda.java.model.Aluno;
import io.github.henriquemcc.agenda.java.ui.adapter.ListaAlunosAdapter;

public class ListaAlunosView
{

	public void confirmaRemocao(@NonNull final MenuItem item)
	{
		new AlertDialog.Builder(this).setTitle("Removendo aluno").setMessage("Tem certeza que quer remover o aluno?").setPositiveButton("Sim", (dialogInterface, i) ->
		{
			final AdapterView.AdapterContextMenuInfo menuInfo = (AdapterView.AdapterContextMenuInfo) item.getMenuInfo();
			final Aluno alunoEscolhido = (Aluno) adapter.getItem(menuInfo.position);
			remove(alunoEscolhido);
		}).setNegativeButton("Não", null).show();
	}

	public void atualizaAlunos()
	{
		adapter.atualiza(dao.todos());
	}

	public void remove(Aluno aluno)
	{
		dao.remove(aluno);
		adapter.remove(aluno);
	}

	public void configuraAdapter(ListView listaDeAlunos)
	{
		adapter = new ListaAlunosAdapter(this);
		listaDeAlunos.setAdapter(adapter);
	}
}

File: Kotlin/Agenda/app/src/main/java/io/github/henriquemcc/agenda/kotlin/ui/ListaAlunosView.kt
package io.github.henriquemcc.agenda.kotlin.ui

import android.system.Os.remove
import android.view.MenuItem
import android.widget.AdapterView
import android.widget.ListView
import androidx.appcompat.app.AlertDialog
import io.github.henriquemcc.agenda.kotlin.model.Aluno
import io.github.henriquemcc.agenda.kotlin.ui.adapter.ListaAlunosAdapter

class ListaAlunosView {

    fun confirmaRemocao(item: MenuItem) {
        AlertDialog.Builder(this)
            .setTitle("Removendo aluno")
            .setMessage("Tem certeza que quer remover o aluno?")
            .setPositiveButton("Sim") { _, _ ->
                val menuInfo = item.menuInfo as AdapterView.AdapterContextMenuInfo
                val alunoEscolhido = adapter.getItem(menuInfo.position)
                remove(alunoEscolhido)
            }
            .setNegativeButton("Não", null)
            .show()
    }

    fun atualizaAlunos()
    {
        adapter.atualiza(dao.todos())
    }

    fun remove(aluno: Aluno)
    {
        dao.remove(aluno)
        adapter.remove(aluno)
    }

    fun configuraAdapter(listaDeAlunos: ListView?)
    {
        adapter = ListaAlunosAdapter(this)
        listaDeAlunos?.adapter = adapter
    }
}